dhcp-protocol/src/main/java/org/anarres/dhcp/v6/io/Dhcp6RequestContext.java

```diff
@@ -0,0 +1,61 @@
+package org.anarres.dhcp.v6.io;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import java.net.InetAddress;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.anarres.dhcp.v6.options.Dhcp6Options;
+
+/**
+ * Request context with IP information about the client
+ */
+public class Dhcp6RequestContext {
+
+    private final InetAddress clientAddress;
+    private final InetAddress linkAddress;
+    private final Optional<Dhcp6Options> relayedOptions;
+
+    /**
+     * Non-relayed messages
+     */
+    public Dhcp6RequestContext(@Nonnull final InetAddress clientAddress) {
+        this(clientAddress, clientAddress, null);
+    }
+
+    /**
+     * Relayed messages
+     */
+    public Dhcp6RequestContext(@Nonnull final InetAddress linkAddress, @Nonnull final InetAddress clientAddress,
+        @Nullable final Dhcp6Options options) {
```

Comment:
Given that Dhcp6Options is a container, would a simpler contract be that the container is nonnull but possibly empty, rather than wrapping the container in an Optional?